pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -99,8 +103,41 @@ public boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnM
       return false;
     }
 
-    Comparable value = getValue(filterQueryTree.getValue().get(0), metadata.getDataType());
     PartitionFunction partitionFunction = metadata.getPartitionFunction();
+
+    if (operator == FilterOperator.EQUALITY) {
+      return pruneEquality(filterQueryTree, metadata, partitionRanges, partitionFunction);
+    }
+
+    return pruneIn(filterQueryTree, metadata, partitionRanges, partitionFunction);
```

Comment:
There are more than necessary 'if' statements here. And it will complicate the code as more predicates are added. Better to have just one switch-case so that there's no more than one branching per predicate.